services/monarch-link-app/app.py

```diff
@@ -44,49 +51,127 @@ def read_data():
         try:
             return json.load(openfile)
         except Exception:
+            # return a dict if the file is empty
             return dict()
 
 
-@app.route("/create/<id>", methods=["POST"])
+# generate an id that does not already exist
+def generate_code(svgData):
+    code = ''.join([str(random.randint(1, 9)) for i in range(6)])
+    while code in svgData:
+        code = generate_code(svgData)
+    return code
+
+
+# custom converter to validate that the id
+# in the url is a valid id
+class CodeConverter(BaseConverter):
+    def to_python(self, value):
+        # has six digits between 1 and 8
+        pattern = re.compile("[1-8]{6}")
+        # also has a length of six
+        if not pattern.match(value) or len(value) != 6:
+            logging.debug('Received request with invalid ID value')
+            raise ValidationError('Invalid id value')
+        return value
+
+    def to_url(self, value):
+        return value
+
+
+app.url_map.converters['code'] = CodeConverter
+
+
+@app.route("/create/<string:title>", methods=["POST"])
```

Comment:
I wouldn't have put the title as part of the path here. This makes more sense as part of the request body.